spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -874,6 +990,43 @@ public String toString() {
 
 		}
 
+		private final class ConsumerBatchAcknowledgment implements Acknowledgment {
+
+			private final List<ConsumerRecord<K, V>> records;
+
+			private final boolean immediate;
+
+			private ConsumerBatchAcknowledgment(List<ConsumerRecord<K, V>> records, boolean immediate) {
+				// make a copy in case the listener alters the list
+				this.records = new LinkedList<ConsumerRecord<K, V>>(records);
```

Comment:
That is precisely a case. See `FilteringBatchMessageListenerAdapter`.
But we have to ack those records anyway.
Correct?